### DISCUSSION TITLE
API call for current song cover

### DISCUSSION THREAD

**Discussion by payback007** [jcorporation/myMPD#555] [Q&A] [answered]
myMPD version: [8.0.4]

Just updated to the new version! Thank you very much for the great work. After I could implement a few instances, which are quite working good.

But I have a problem with API call of the cover image: I call the cover image with following code:

`URI=$(curl -s -k http://IP:PORT/api/ -d'{"jsonrpc":"2.0","id":0,"method":"MYMPD_API_PLAYER_CURRENT_SONG","params":{}}' | jq -r .result.images)
echo "http://IP:PORT/albumart/${URI}"`

The problem is, the answer of the API call is unfortunately with some signs I didn't expect:

`http://IP:PORT/albumart/[" path_to_the_current_cover"]`

How to remove the "[" and "]" signs?

**Comment by jcorporation** [comment]
result.images is an array of all images myMPD has  found. 

`jq -r ".result.images | .[1]"` should select the first array element.

But it’s easier:
To get the coverimage you can simply call `/albumart/<song uri>`

**Comment by payback007** [comment]
Thank you, it is working now with this call:

URI=$(curl -s -k http://IP:PORT/api/ -d'{"jsonrpc":"2.0","id":0,"method":"MYMPD_API_PLAYER_CURRENT_SONG","params":{}}' | jq -r ".result.images | .[]")

Unfortunately I'm not so familiar with API calls, how would the simple was be? In the API documentation I couldn't find something about "song uri".

**Comment by jcorporation** [comment]
MYMPD_API_PLAYER_CURRENT_SONG should return the song uri. Use this uri and prepend /albumart/

**Comment by payback007** [comment]
Yes you are right, but the song-uri has as extension the file type like .m4a/.mp3 etc. -> with that I can't deal, I need the path to the cover image. But doesn't matter, it seems to work.

**Reply by jcorporation** [reply]
The extension of the requested file has nothing todo with the responded content mime type.

For the song uri `Alben/Monolith/Compressed_Form/02.Decompressed_Form.mp3` the following request gets the albumart:
`https://localhost/albumart/Alben/Monolith/Compressed_Form/02.Decompressed_Form.mp3`

With this call myMPD searches for your coverimage. You need only to know the song-uri.

**Comment by jcorporation** [comment, marked as answer]
https://github.com/jcorporation/myMPD/discussions/557#discussioncomment-1250661
